lte/gateway/python/magma/policydb/servicers/session_servicer.py

```diff
@@ -124,74 +133,75 @@ def TerminateSession(
-        """
-        return PolicyRule(
-            # Don't set the rating group
-            # Don't set the monitoring key
-            # Don't set the hard timeout
-            id=policy_id,
-            priority=2,
-            flow_list=self._get_allow_all_flows(),
-            tracking_type=PolicyRule.TrackingType.Value("NO_TRACKING"),
-        )
-
-    def _get_allow_all_flows(self) -> List[FlowDescription]:
-        """
-        Returns:
-            Two flows, for outgoing and incoming traffic
+        Get a list of dynamic rules to install
+        Currently only includes a single rule for allow-all of traffic
         """
         return [
-            # Set flow match for ll packets
-            # Don't set the app_name field
-            FlowDescription(  # uplink flow
-                match=FlowMatch(
-                    direction=FlowMatch.Direction.Value("UPLINK"),
-                ),
-                action=FlowDescription.Action.Value("PERMIT"),
-            ),
-            FlowDescription(  # downlink flow
-                match=FlowMatch(
-                    direction=FlowMatch.Direction.Value("DOWNLINK"),
-                ),
-                action=FlowDescription.Action.Value("PERMIT"),
-            ),
+            DynamicRuleInstall(
+                policy_rule=get_allow_all_policy_rule(subscriber_id, apn)),
         ]
 
-    def _get_rules_for_imsi(self, imsi: str) -> List[StaticRuleInstall]:
+    def _get_session_static_rules(
+        self,
+        imsi: str,
+        apn: str,
+    ) -> List[StaticRuleInstall]:
         """
         Get the list of static rules to be installed for a subscriber
         NOTE: Remove "IMSI" prefix from imsi argument.
         """
-        try:
-            info = self._subscriberdb_stub.GetSubscriberData(NetworkID(id=imsi))
-            return [StaticRuleInstall(rule_id=rule_id)
-                    for rule_id in info.lte.assigned_policies]
-        except grpc.RpcError:
-            logging.error('Unable to find data for subscriber %s', imsi)
+        if imsi not in self._apn_rules_by_sid:
             return []
 
+        sub_apn_policies = self._apn_rules_by_sid[imsi]
+        assigned_static_rules = [] # type: List[StaticRuleInstall]
+        # Add global rules
+        global_rules = self._get_global_static_rules(sub_apn_policies)
+        assigned_static_rules += \
+            list(map(lambda id: StaticRuleInstall(rule_id=id),
+                     global_rules))
+        # Add APN specific rules
+        for apn_policy_set in sub_apn_policies.rules_per_apn:
+            if apn_policy_set.apn != apn:
+                continue
+            # Only add rules if the APN matches
+            static_rule_ids = self._get_static_rules(apn_policy_set)
+            assigned_static_rules +=\
+                list(map(lambda id: StaticRuleInstall(rule_id=id),
+                         static_rule_ids))
+
+        return assigned_static_rules
+
+    def _get_global_static_rules(
```

Comment:
These two functions below look almost the same, should we turn this into a single function that takes a repeated policy_name and a repeated base_name?